File: android/src/main/java/com/reactnativecarousel/CarouselViewManager.kt
package com.reactnativecarousel

import com.facebook.react.uimanager.PixelUtil
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.ViewGroupManager
import com.facebook.react.uimanager.annotations.ReactProp


class CarouselViewManager : ViewGroupManager<CarouselView>() {
  override fun getName() = "CarouselView"

  override fun createViewInstance(reactContext: ThemedReactContext): CarouselView {
    return CarouselView(reactContext)
  }

  @ReactProp(name = "nextItemVisible")
  fun nextItemVisible(view: CarouselView, value: Int) {
    view.nextItemVisiblePx = PixelUtil.toPixelFromDIP(value.toFloat())
  }

  @ReactProp(name = "currentItemHorizontalMargin")
  fun currentItemHorizontalMargin(view: CarouselView, value: Int) {
    view.currentItemHorizontalMarginPx = PixelUtil.toPixelFromDIP(value.toFloat())
  }
}

File: ios/CarouselViewManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(CarouselView, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(currentItemHorizontalMargin, NSNumber);
RCT_EXPORT_VIEW_PROPERTY(nextItemVisible, NSNumber);
RCT_EXPORT_VIEW_PROPERTY(pageSelect, RCTBubblingEventBlock)
@end
